### PR TITLE
Mongodb adapter fixes + scanner

### PR DESCRIPTION
**Pull request by zeroX-tj**
This works for me. The config still needs to be fixed as well as some config values for the exchange are not there when running from the ui. So i had to add a workaround for now.

### CODE REVIEW

File: core/markets/backtest.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ var dirs = util.dirs();
 var log = require(dirs.core + 'log');
 var moment = require('moment');
 
-var adapter = config[config.adapter];
+var adapter = config.adapters[config.adapter];
```
Comment (askmike) [inline core/markets/backtest.js]: The config format has actually changed, you changed it to the old format.

File: core/tools/dateRangeScanner.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ var config = util.getConfig();
 var dirs = util.dirs();
 var log = require(dirs.core + 'log');
 
-var adapter = config[config.adapter];
+var adapter = config.adapters[config.adapter];
```
Comment (askmike) [inline core/tools/dateRangeScanner.js]: dito

File: plugins/mongodb/scanner.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    this.db = handle;
+
+    let markets = [];
+    async.waterfall([
+        (cb) => {
+            handle.getCollectionNames(cb)
+        },
+        (collections, cb) => {
+            async.each(collections, (collection, cb) => {
+                let [exchange, type] = collection.split('_');
```
Comment (askmike) [inline plugins/mongodb/scanner.js:18]: Do you know if node v6.0 (minimum as required by Gekko) supports this?

**Review by askmike — CHANGES_REQUESTED** [review]

File: plugins/mongodb/scanner.js (hunk 3 above)
Comment (zeroX-tj) [reply plugins/mongodb/scanner.js:18]: Works for 6.0 (tested)

**Comment by askmike** [comment]
👍  I'll ping you if there are issues with mongo at some point.

**Comment by askmike** [comment]
thanks a lot :)

**Pull request merged**